core/arch/arm/mm/core_mmu.c

```diff
@@ -994,6 +994,101 @@ static void set_pg_region(struct core_mmu_table_info *dir_info,
 	}
 }
 
+TEE_Result core_mmu_map_pages(vaddr_t vstart, paddr_t *pages, size_t num_pages,
+			      enum teecore_memtypes memtype)
+{
+	TEE_Result ret;
+	struct core_mmu_table_info tbl_info;
+	struct tee_mmap_region *mm;
+	unsigned int idx;
+	uint32_t old_attr;
+	vaddr_t vaddr = vstart;
+	size_t i;
+
+	if (vaddr & SMALL_PAGE_MASK)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	mm = find_map_by_va((void *)vaddr);
+
+	if (!mm)
+		panic("VA does not belong to any known mm region");
```

Comment:
@jforissier in general I'd agree, but mapping memory is a bit of a special case. Here prefer `panic()` or `assert()` over error codes because there's usually no means of recovery.